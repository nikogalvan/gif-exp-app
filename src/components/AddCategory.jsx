/* eslint-disable react/prop-types */
import { useState } from "react";

export const AddCategory = ({ onAddGif }) => {
  const [inputValue, setInputValue] = useState("");
  const onChangeHandler = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmitHandler = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    onAddGif(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmitHandler}>
      <input
        type="text"
        value={inputValue}
        onChange={onChangeHandler}
        placeholder="Buscar gifs"
      />
    </form>
  );
};
