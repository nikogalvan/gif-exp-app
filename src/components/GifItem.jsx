import PropTypes from "prop-types";

export const GifItem = ({ title, url, id }) => {
  return (
    <div className="card">
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.number
};
