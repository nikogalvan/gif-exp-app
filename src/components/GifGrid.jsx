import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}
      <div>
        {gifs.map((gif) => (
          <GifItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};
