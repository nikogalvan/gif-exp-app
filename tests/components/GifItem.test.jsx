/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en <GifItem /> ", () => {
  const title = "Hero";
  const url =
    "https://media4.giphy.com/media/v1.Y2lkPWQwNDZhZDAxMHBjbm51Mng0emhsdm9pYmNseWQwd3NjZnhkajgxYWN4bG50NzVheSZlcD12MV9naWZzX3NlYXJjaCZjdD1n/arbHBoiUWUgmc/giphy.gif";

  test("debe hacer match con el snapshot ", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debee de mostrar la imagen con el URL y el ALT indicado", () => {
    // Renderiza el componente GifItem con las props title y url.
    render(<GifItem title={title} url={url} />);

    // Alternativa para depuración: Puedes usar screen.debug() para inspeccionar el contenido renderizado.
    // screen.debug();

    // Alternativa: Probar directamente las propiedades del elemento img obteniéndolo por su rol.
    // expect(screen.getByRole("img").src).toBe(url); // Verifica que la propiedad src sea igual a url.
    // expect(screen.getByRole("img").alt).toBe(title); // Verifica que la propiedad alt sea igual a title.

    // Destructuración: Obtiene las propiedades src y alt del elemento img.
    const { src, alt } = screen.getByRole("img");

    // Comprueba que src coincide con la URL proporcionada.
    expect(src).toBe(url);

    // Comprueba que alt coincide con el título proporcionado.
    expect(alt).toBe(title);
  });

  test("debe mostrar el titulo en el component", () => {
    render(<GifItem title={title} url={url} />);
    //console.log(screen.getByText(title));

    expect(screen.getByText(title)).toBeTruthy();
  });
});
