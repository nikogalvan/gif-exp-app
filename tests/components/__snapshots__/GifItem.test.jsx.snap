// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem />  debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Hero
    </p>
    <img
      alt="Hero"
      src="https://media4.giphy.com/media/v1.Y2lkPWQwNDZhZDAxMHBjbm51Mng0emhsdm9pYmNseWQwd3NjZnhkajgxYWN4bG50NzVheSZlcD12MV9naWZzX3NlYXJjaCZjdD1n/arbHBoiUWUgmc/giphy.gif"
    />
  </div>
</div>
`;
